### DISCUSSION TITLE
Keybindings for control center not present after upgrade to Kinetic

### DISCUSSION THREAD

**Discussion by hnrkgrln** [regolith-linux/regolith-desktop#794] [Q&A] [answered]
I just upgraded to Kinetic and for some reason the keybindings (mod+c and the others) for the control center aren't working anymore. regolith-control-center is installed and I can open it from cmdline.

Is there a package that I am missing perhaps?

**Comment by kgilmer** [comment]
Hi  @hnrkgrln , it might be helpful to see which regolith packages you have installed.  You can get a list with `apt list --installed | grep regolith`.  The [i3 config package](https://regolith-desktop.com/docs/using-regolith/configuration/#all-i3-configuration-packages) that adds keybindings to settings is ~~`regolith-i3-gnome`~~ `regolith-i3-control-center-regolith` or `regolith-i3-control-center-gnome`.  If there is some error happening you may find it in `/var/log/syslog`.

**Reply by dmikushin** [reply]
Hi @kgilmer , thank you for Regolith, it's where I finally found my peace. I confirm that for me some keybindings have recently stopped working, in the middle of nothing on 22.04, I didn't even make any upgrade. Only a few keybindings stopped working, such as Mod+d (displays), which I frequently use. Control center keybinding is also lost. Interestingly, they also disappeared from Mod+Shift+? index. I couldn't delete them I'm sure, I have no idea how, my setup is super-basic. Let me offer a screenshot of native Regolith help screen versus another "good" Regolith 20.04 in a VNC session:

![Screenshot from 2022-12-19 21-00-58](https://user-images.githubusercontent.com/5672874/208510265-96ded189-1ed9-4767-a4c1-9c4f504afa91.png)

**Reply by dmikushin** [reply]
What you asked:

```
apt list --installed | grep regolith                                                                                              832.259s 21:05

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

arc-icon-theme/unknown,now 20161122-1-1regolith amd64 [installed,automatic]
ayu-theme/unknown,now 0.2.2-1-1regolith amd64 [installed,automatic]
fonts-nerd-font-bitstreamverasansmono/unknown,now 2.1.0-2-1regolith all [installed,automatic]
fonts-nerd-font-hack/unknown,now 2.1.0-2-1regolith all [installed,automatic]
fonts-nerd-font-meslo/unknown,now 2.1.0-2-1regolith all [installed,automatic]
gruvbox-gtk/unknown,now 1.0.1-1-1regolith amd64 [installed,automatic]
i3-gaps-wm/unknown,now 4.20.1-1regolith amd64 [installed,automatic]
i3-next-workspace/unknown,now 1.0.1-1-1regolith amd64 [installed,automatic]
i3xrocks-app-launcher/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-battery/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-cpu-usage/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
i3xrocks-focused-window-name/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-info/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-memory/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-net-traffic/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
i3xrocks-rofication/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-time/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks/unknown,now 1.3.5-1-1regolith amd64 [installed,automatic]
ilia/unknown,now 0.12.14-jammy-1regolith amd64 [installed,automatic]
picom/unknown,now 9.0.0-1regolith amd64 [installed,automatic]
regolith-compositor-picom-glx/unknown,now 1.2.4-1regolith amd64 [installed]
regolith-control-center/unknown,now 1.41.18-1regolith amd64 [installed,automatic]
regolith-default-settings/unknown,now 1.0.2-1regolith amd64 [installed,automatic]
regolith-desktop/unknown,now 4.2.0-1regolith amd64 [installed]
regolith-ftue/unknown,now 2.1.5-1regolith amd64 [installed,automatic]
regolith-i3-base-launchers/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-compositor/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-config/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-default-style/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-ftue/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-gaps-partial/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-i3xrocks/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-ilia/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-navigation/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-next-workspace/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-resize/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-rofication-ilia/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-session/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-unclutter/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-user-programs/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-workspace-config/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3xrocks-config/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
regolith-look-ayu-dark/unknown,now 0.8.4-1regolith amd64 [installed]
regolith-look-default/unknown,now 0.7.1-1regolith amd64 [installed,automatic]
regolith-look-solarized-dark/unknown,now 0.8.4-1regolith amd64 [installed]
regolith-rofication/unknown,now 1.4.2-1-1regolith amd64 [installed,automatic]
regolith-session-flashback/unknown,now 0.5.0-1regolith amd64 [installed,automatic]
regolith-unclutter-xfixes/unknown,now 1.5-2-1regolith amd64 [installed,automatic]
remontoire/unknown,now 1.4.2-1regolith amd64 [installed,automatic]
solarc-theme/unknown,now 800c997-2-1regolith amd64 [installed,automatic]
xrescat/unknown,now 1.2.1-1-1regolith amd64 [installed,automatic]
```

**Reply by dmikushin** [reply]
I actually have another 22.04, where Mod+d and Mod+c are still working. ~~Currently I have Regolith on 4 laptops, and 3 VNCs.~~ Here is the output:

```
apt list --installed | grep regolith

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

fonts-nerd-font-bitstreamverasansmono/unknown,now 2.1.0-2-1regolith all [installed,automatic]
i3-gaps-wm/unknown,now 4.20.1-1regolith amd64 [installed,automatic]
i3-next-workspace/unknown,now 1.0.1-1-1regolith amd64 [installed,automatic]
i3xrocks-app-launcher/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-battery/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-cpu-usage/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
i3xrocks-focused-window-name/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-info/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-memory/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-net-traffic/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
i3xrocks-rofication/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-time/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
i3xrocks/unknown,now 1.3.5-1-1regolith amd64 [installed,automatic]
ilia/unknown,now 0.12.14-jammy-1regolith amd64 [installed,automatic]
regolith-compositor-none/unknown,now 1.0.3-1-1regolith amd64 [installed,automatic]
regolith-control-center/unknown,now 1.41.18-1regolith amd64 [installed,automatic]
regolith-default-settings/unknown,now 1.0.2-1regolith amd64 [installed,automatic]
regolith-desktop/unknown,now 4.2.0-1regolith amd64 [installed]
regolith-ftue/unknown,now 2.1.5-1regolith amd64 [installed,automatic]
regolith-i3-base-launchers/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-compositor/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-config/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-default-style/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-ftue/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-gaps-partial/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-gnome/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-i3xrocks/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-ilia/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-navigation/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-networkmanager/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-next-workspace/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-resize/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-rofication-ilia/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-session/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-unclutter/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-user-programs/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-workspace-config/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3xrocks-config/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
regolith-look-default/unknown,now 0.7.1-1regolith amd64 [installed,automatic]
regolith-rofication/unknown,now 1.4.2-1-1regolith amd64 [installed,automatic]
regolith-session-flashback/unknown,now 0.5.0-1regolith amd64 [installed,automatic]
regolith-unclutter-xfixes/unknown,now 1.5-2-1regolith amd64 [installed,automatic]
remontoire/unknown,now 1.4.2-1regolith amd64 [installed,automatic]
xrescat/unknown,now 1.2.1-1-1regolith amd64 [installed,automatic]
```

**Reply by dmikushin** [reply]
As you can see they are identical, except some extra styles and compositor. But I'm sure styles were installed long ago, and compositor is installed recently, when keybindings were already broken. I think the issue was introduced by some conflicting update of other packages, could it be that GNOME is hitting us in the back?

**Reply by kgilmer** [reply]
~~The key difference between these two systems is that the former lacks `regolith-i3-gnome`.  I believe this is the root cause of why the keybindings were not present.  Next is to solve the mystery of why `regolith-i3-gnome` was not installed..~~

I'd forgotten a change in 2.2 regarding i3 <-> settings integration, and as a result was looking in the wrong place to root-cause the issue.  My apologies to you and others (@hnrkgrln, @winterec)  impacted by this bug.  Here is the story:

In Regolith 2.2 the package `regolith-i3-gnome` was deprecated.  This package contained keybindings for `regolith-control-center`.  Some people want to run Regolith on unsupported Ubuntu variants that could not install `regolith-control-center` due to shipping different versions of GNOME than baseline Ubuntu.  As the project does not have resources to expand testing to other Ubuntu variants at this time, I opted to [create two new packages](https://github.com/regolith-linux/regolith-i3-config/commit/321220f5e6e047c5710e8da3366108ebdaa95a12) `regolith-i3-control-center-regolith` and  `regolith-i3-control-center-gnome` which launch either regolith or upstream versions of `settings` respectively.  

I'm guessing that for users impacted, during upgrade, the deprecated package was removed but neither of the new ones were installed, leading to the missing keybindings.  Additionally, both of these packages are "soft" dependencies meaning they are not required for regolith to function.

In summary, for i3 keybindings that launch upstream settings (`gnome-control-center`), install the `regolith-i3-control-center-gnome` package, and for regolith settings, install `regolith-i3-control-center-regolith`.

**Reply by dmikushin** [reply]
I'm 100% sure it worked initially from the fresh start of the system, because I use Mod+d many times everyday. So the only cause I can imagine is that it was removed by apt-get by some other package installation.

**Reply by dmikushin** [reply]
Ah, I remember now! It was [package `qtbase5-private-dev` is trying to remove most of the desktop system on Ubuntu 22.04](https://forum.qt.io/topic/137839/issues-installing-qtbase5-private-dev-on-an-ubuntu-22-04-system). I let it run maybe for a couple of seconds, which was enough to destroy `regolith-i3-gnome`. Now I have a nice story for a water-cooler conversation!

**Comment by hnrkgrln** [comment]
This is interesting. But first, these are my installed packages.
```
$ apt list --installed | grep regolith

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

arc-icon-theme/unknown,now 20161122-1-1regolith amd64 [installed,automatic]
ayu-theme/unknown,now 0.2.2-1-1regolith amd64 [installed,automatic]
dracula-gtk/unknown,now 1.1.1-1regolith amd64 [installed,automatic]
fonts-nerd-font-bitstreamverasansmono/unknown,now 2.1.0-2-1regolith all [installed,automatic]
fonts-nerd-font-cousine/unknown,now 2.1.0-2-1regolith all [installed,automatic]
fonts-nerd-font-dejavusansmono/unknown,now 2.1.0-2-1regolith all [installed]
fonts-nerd-font-hack/unknown,now 2.1.0-2-1regolith all [installed,automatic]
fonts-nerd-font-meslo/unknown,now 2.1.0-2-1regolith all [installed,automatic]
fonts-nerd-font-mononoki/unknown,now 2.1.0-2-1regolith all [installed,automatic]
fonts-nerd-font-robotomono/unknown,now 2.1.0-2-1regolith all [installed,automatic]
fonts-nerd-font-sourcecodepro/unknown,now 2.1.0-2-1regolith all [installed,automatic]
gruvbox-gtk/unknown,now 1.0.1-1-1regolith amd64 [installed,automatic]
i3-gaps-wm/unknown,now 4.21-2-1regolith amd64 [installed,automatic]
i3-next-workspace/unknown,now 1.0.2-1regolith amd64 [installed,automatic]
i3-swap-focus/unknown,now 0.4.3-1regolith amd64 [installed,automatic]
i3xrocks-battery/unknown,now 5.4.6-1regolith amd64 [installed]
i3xrocks-bluetooth/unknown,now 5.4.6-1regolith amd64 [installed]
i3xrocks-cpu-usage/unknown,now 5.4.6-1regolith amd64 [installed,automatic]
i3xrocks-memory/unknown,now 5.4.6-1regolith amd64 [installed]
i3xrocks-net-traffic/unknown,now 5.4.6-1regolith amd64 [installed,automatic]
i3xrocks-rofication/unknown,now 5.4.6-1regolith amd64 [installed]
i3xrocks-time/unknown,now 5.4.6-1regolith amd64 [installed]
i3xrocks-volume/unknown,now 5.4.6-1regolith amd64 [installed]
i3xrocks/unknown,now 1.3.5-1-1regolith amd64 [installed,automatic]
ilia/unknown,now 0.12.29-jammy-1regolith amd64 [installed,automatic]
nordic/unknown,now 2.1.0-1-1regolith all [installed,automatic]
plymouth-theme-regolith-logo/unknown,now 1.2.1-1regolith all [installed]
regolith-compositor-picom-glx/unknown,now 1.2.4-1regolith amd64 [installed]
regolith-control-center/unknown,now 1.43.1-3-1regolith amd64 [installed]
regolith-default-settings/unknown,now 2.0.4-1regolith amd64 [installed,automatic]
regolith-desktop/unknown,now 4.3.0-1regolith amd64 [installed]
regolith-ftue/unknown,now 2.1.6-1regolith amd64 [installed,automatic]
regolith-i3-base-launchers/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-compositor/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-config/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-default-style/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-ftue/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-gaps-partial/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-i3xrocks/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-ilia/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-navigation/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-networkmanager/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-next-workspace/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-resize/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-rofication-ilia/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-session/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-swap-focus/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-unclutter/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-user-programs/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3-workspace-config/unknown,now 3.8.1-jammy-1regolith amd64 [installed,automatic]
regolith-i3xrocks-config/unknown,now 5.4.6-1regolith amd64 [installed,automatic]
regolith-lightdm-config/unknown,now 1.3.5-1regolith amd64 [installed]
regolith-look-ayu-dark/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-look-ayu-mirage/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-look-ayu/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-look-blackhole/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-look-default-loader/unknown,now 0.7.5-1regolith amd64 [installed,automatic]
regolith-look-default/unknown,now 0.7.5-1regolith amd64 [installed]
regolith-look-dracula/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-look-gruvbox/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-look-i3-default/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-look-lascaille/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-look-nevil/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-look-nord/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-look-solarized-dark/unknown,now 0.8.12-1regolith amd64 [installed]
regolith-rofication/unknown,now 1.4.2-1-1regolith amd64 [installed,automatic]
regolith-session-flashback/unknown,now 0.6.5-1regolith amd64 [installed,automatic]
regolith-unclutter-xfixes/unknown,now 1.5-2-1regolith amd64 [installed,automatic]
solarc-theme/unknown,now 800c997-2-1regolith amd64 [installed,automatic]
whitesur-gtk-theme/unknown,now 1.0-2202.02.21-1regolith amd64 [installed,automatic]
xrescat/unknown,now 1.2.1-2-1regolith amd64 [installed,automatic]
```

This is what I get when i try to install regolith-i3-gnome package:
```
$ sudo apt install regolith-i3-gnome
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package regolith-i3-gnome is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  regolith-i3-control-center-regolith regolith-i3-control-center-gnome

E: Package 'regolith-i3-gnome' has no installation candidate
```

Is it a problem with the apt repos?

EDIT:
Oh, and syslog gave me nothing useful.

**Reply by dmikushin** [reply]
Looks like your update hasn't finished, most of the system is not Kinetic yet.

**Comment by hnrkgrln** [comment, marked as answer]
Okay, so I did this. And it works. 
```
$ sudo apt install regolith-i3-control-center-gnome
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libavahi-ui-gtk3-0 network-manager-config-connectivity-ubuntu
Use 'sudo apt autoremove' to remove them.
The following NEW packages will be installed:
  regolith-i3-control-center-gnome
0 upgraded, 1 newly installed, 0 to remove and 2 not upgraded.
Need to get 9 034 B of archives.
After this operation, 21,5 kB of additional disk space will be used.
Get:1 https://regolith-desktop.org/release-ubuntu-kinetic-amd64 kinetic/main amd64 regolith-i3-control-center-gnome amd64 3.8.1-jammy-1regolith [9 034 B]
Fetched 9 034 B in 4s (2 548 B/s)                                
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package regolith-i3-control-center-gnome.
(Reading database ... 360137 files and directories currently installed.)
Preparing to unpack .../regolith-i3-control-center-gnome_3.8.1-jammy-1regolith_amd64.deb ...
Unpacking regolith-i3-control-center-gnome (3.8.1-jammy-1regolith) ...
Setting up regolith-i3-control-center-gnome (3.8.1-jammy-1regolith) ...
```
But it says jammy. So I guess something is up with the repos?

**Reply by dmikushin** [reply]
I don't have `regolith-i3-control-center-gnome` on Jammy.

**Reply by hnrkgrln** [reply]
Yes, something is not right. I will have to look into that. But mod+c is working now so I guess this is solved.

**Reply by kgilmer** [reply]
To clarify, the packages `regolith-i3-control-center-gnome` and `regolith-i3-control-center-regolith` were introduced in Regolith 2.2 and replaced/deprecated `regolith-i3-gnome`.  That the package `regolith-i3-control-center-gnome` has "jammy" in the version is (while admittedly confusing) not a bug.  In Regolith some packages need to maintain different branches due to dependency differences across different versions of Ubuntu.  The convention is to create a branch for the variant with it's name.  In this case `jammy` introduced a change requiring this repo to be branched.  The source of truth regarding which "variant" of Regolith gets installed on a system is the "codename" in `/etc/apt/sources.list.d/regolith.list`. Example from my system, here the codename is "testing" (debian bookworm):

```
deb [arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] https://regolith-desktop.org/unstable-debian-testing-amd64 testing main
```

**Reply by hnrkgrln** [reply]
Great! Thank you for the clarification. 

I had some other issues after upgrading as well, so it wasn't clear to me if it had to do with Regolith or a messy upgrade in general.

Keep up the good work!

**Comment by dmikushin** [comment]
OK, it works for me too now!! What I did on Jammy:

```
sudo apt update
sudo apt install regolith-i3-control-center-gnome
regolith-look refresh
```
